app/controllers/api/v1/orders_controller.rb

```diff
@@ -0,0 +1,20 @@
+class Api::V1::OrdersController < ApplicationController
+  require_dependency "#{Rails.root}/app/validators/orders_controller"
+
+  before_action :validate_params, only: :index
+
+  def index
+    @orders = Order.within_range(params[:from_date], params[:to_date])
+    json_response(@orders)
+  end
+
+  private
+
+  def validate_params
+    validator = Validators::OrdersController.new(params)
+
+    return if validator.valid?
+
+    json_response({ errors: validator.errors }, :unprocessable_entity)
+  end
```

Comment:
Here I decided to go with a **validator class** as I believe it lends itself to easier parameter validation since we can leverage `ActiveModel`'s validation helpers.

I've considered attempting to place a similar sort of structure here but felt that wasn't really a very scalable solution.

I also considered creating a custom validator from scratch (without leveraging `ActiveModel`'s paradigm) but felt this was the simpler and cleaner approach. Also as a bonus it wouldn't be hard in for other devs to learn understand this validation pattern.